permissions_validators/src/private_blockchain/mod.rs

```diff
@@ -38,3 +38,541 @@ impl<W: WorldTrait> IsGrantAllowed<W> for ProhibitGrant {
+        bronze_asset_id: AssetId,
+        bronze_asset_definition_id: AssetDefinitionId,
+        first_domain: (DomainId, Domain),
+        second_domain: (DomainId, Domain),
+        wsv: WorldStateView<World>,
+    }
+
+    impl TestEnv {
+        fn new() -> Self {
+            // Alice
+            let alice_id = AccountId::from_str("alice@test0").expect("Valid");
+            let mut alice = Account::new(alice_id.clone(), []).build();
+
+            // Gold asset
+            let gold_asset_definition_id =
+                AssetDefinitionId::from_str("gold#test0").expect("Valid");
+            let gold_asset_id = AssetId::new(gold_asset_definition_id.clone(), alice_id.clone());
+            let gold_asset = Asset::new(gold_asset_id.clone(), AssetValue::Quantity(100));
+
+            // Alice owns Gold
+            alice.add_asset(gold_asset);
+
+            // Bob
+            let bob_id = AccountId::from_str("bob@test0").expect("Valid");
+            let mut bob = Account::new(bob_id.clone(), []).build();
+
+            // Silver asset
+            let silver_asset_definition_id =
+                AssetDefinitionId::from_str("silver#test0").expect("Valid");
+            let silver_asset_id = AssetId::new(silver_asset_definition_id.clone(), bob_id.clone());
+            let silver_asset = Asset::new(silver_asset_id.clone(), AssetValue::Quantity(200));
+
+            // Bob owns Silver
+            bob.add_asset(silver_asset);
+
+            // Carol
+            let carol_id = AccountId::from_str("carol@test1").expect("Valid");
+            let mut carol = Account::new(carol_id.clone(), []).build();
+
+            // Bronze asset
+            let bronze_asset_definition_id =
+                AssetDefinitionId::from_str("bronze#test1").expect("Valid");
+            let bronze_asset_id =
+                AssetId::new(bronze_asset_definition_id.clone(), carol_id.clone());
+            let bronze_asset = Asset::new(bronze_asset_id.clone(), AssetValue::Quantity(300));
+
+            // Carol owns Bronze
+            carol.add_asset(bronze_asset.clone());
+
+            // Alice also owns Bronze
+            alice.add_asset(bronze_asset);
+
+            // First domain
+            let first_domain_id = DomainId::from_str("test0").expect("Valid");
+            let mut first_domain = Domain::new(first_domain_id.clone()).build();
+
+            // First domain includes Alice
+            first_domain.add_account(alice);
+
+            // First domain includes Bob
+            first_domain.add_account(bob);
+
+            // Second domain
+            let second_domain_id = DomainId::from_str("test1").expect("Valid");
+            let mut second_domain = Domain::new(second_domain_id.clone()).build();
+
+            // Second domain includes Carol
+            second_domain.add_account(carol);
+
+            let wsv = WorldStateView::<World>::new(World::with(
+                [first_domain.clone(), second_domain.clone()],
+                vec![],
```

Comment:
Style point: prefer `Vec::new()` for empty immutable vectors. This conveys to the reader that what you made empty is meant to remain empty.